### ISSUE TITLE
Support: Example docs, next steps...

### ISSUE THREAD

**Issue by TJM** [coreos/go-oidc#288]
I am trying to use this go-oidc v3 in gin-gonic. I started to create "middleware" based on the [examples](https://github.com/coreos/go-oidc/tree/v3/example) included here. This was what I came up with initially: https://github.com/TJM/gin-gonic-oidcauth/tree/basic_insecure.

As the branch name states, that is a basic_insecure example, but it is really not useful for authentication, so...

I then did a bit of research and determined that "nonce" and "state" should definitely not be static strings. It seemed like "state" should be something specific to the client (user's browser), and should thus be stored in the session. The nonce *could* be stored in the session as well, but I found https://pkg.go.dev/github.com/letsencrypt/boulder/nonce which seems to implement "nonce as a service" :p ... and it seems to be working as desired. I am actually just using a nonce for both the nonce and the state. I am not really sure whether thats good or bad :) ... Anyhow, now I am here: https://github.com/TJM/gin-gonic-oidcauth/tree/main ... which I think is a "better" (more secure at least) example, but it is still a contrived example, that dumps a bunch of information into the session, and is able to change from "Hello, world" to "Hello, NAME" ...

Now, I feel like the initial authentication is handled (for the callback), but I am not sure how to handle subsequent requests from the user.

1. What "token" should I store for the user (in the session). The `idToken.AccessToken()` is what I am currently storing, but I am not sure how to validate it. (I guess I could just "assume" that the encrypted/authenticated session has not been tampered with, but that seems bad?)
2. Should I do any token validation on subsequent requests, or just store the "iat" and "exp" (currently storing all claims), and validate that they are still valid.
3. Are there any examples about "refresh" tokens? (renewing the token from the app while the user is still active)

Thanks in advance,
Tommy

**Comment by ericchiang** [comment]
Hey Tommy,

These seem like general questions about OAuth 2.0 security and session management, not necessarily a bug in this library.

Unfortunately, the maintainers of this package generally don't have bandwidth to answer support questions like this. Below are some links that might help getting started:

* https://auth0.com/docs/protocols/state-parameters
* https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html
* https://openid.net/specs/openid-connect-core-1_0.html#NonceNotes

Hope this helps,
Eric

**Comment by TJM** [comment]
Thanks for responding @ericchiang! As per the title of my issue I was asking for support, and that is what you offered. Given the information contained in the documents that you linked (by the way the first and third ones were referenced over the weekend), you have answered several questions that I already solved (state/nonce) that were not part of the example code. The owasp information on sessions is probably the information I will need to solve my issue with validating tokens later in the user's session. I am going with the idea that I have validated the user's token one time, and the user's possession of the session ID is as they say "equivalent" to authentication (temporarily). 

I do propose that the examples should be a little more "secure" by design. The example code says to "not" do this in production, but it doesn't really give any direction or hints about what we "should" do. :) Perhaps this ticket will help someone in the future.

Thanks again,
Tommy
